api-update/common/src/test/scala/uk/gov/nationalarchives/api/update/common/CommonTests.scala

```diff
@@ -0,0 +1,137 @@
+package uk.gov.nationalarchives.api.update.common
+
+import com.nimbusds.oauth2.sdk.token.BearerAccessToken
+import org.mockito.ArgumentMatchers._
+import org.mockito.MockitoSugar
+import org.scalatest.matchers.should.Matchers._
+import sangria.ast.Document
+import sttp.client.{HttpError, Response}
+import sttp.model.StatusCode
+import uk.gov.nationalarchives.api.update.common.TestGraphQLObjects.{Data, TestResponse, Variables}
+import uk.gov.nationalarchives.api.update.common.exceptions.{AuthorisationException, GraphQlException}
+import uk.gov.nationalarchives.tdr.GraphQLClient.Extensions
+import uk.gov.nationalarchives.tdr.error.{GraphQlError, HttpException}
+import uk.gov.nationalarchives.tdr.keycloak.KeycloakUtils
+import uk.gov.nationalarchives.tdr.{GraphQLClient, GraphQlResponse}
+
+import scala.concurrent.Future
+
+class CommonTests extends WiremockTest with MockitoSugar {
+
+    "The send method" should "request a service account token" in {
+      val apiUpdate = ApiUpdate()
+
+      val client = mock[GraphQLClient[Data, Variables]]
+      val document = mock[Document]
+      val keycloakUtils = mock[KeycloakUtils]
+
+      when(keycloakUtils.serviceAccountToken(any[String], any[String]))
+        .thenReturn(Future.successful(new BearerAccessToken("token")))
+      when(client.getResult(any[BearerAccessToken], any[Document], any[Option[Variables]]))
+        .thenReturn(Future.successful(GraphQlResponse(Some(Data(TestResponse())), List())))
+      apiUpdate.send(keycloakUtils, client, document, Variables())
+
+      val expectedId = sys.env("CLIENT_ID")
+      val expectedSecret = sys.env("CLIENT_SECRET")
+
+      verify(keycloakUtils).serviceAccountToken(expectedId, expectedSecret)
+    }
+
+    "The update method" should "call the graphql api with the correct data" in {
```

Comment:
Some of these tests start with "The send method", some are "The update method" and some are just the "The method". Is there any difference between them?

Reply:
No reason other than bad copy pasting/reading skills :smile: They're all for the send method so they all say send now.